server/app.js

```diff
@@ -157,16 +160,17 @@ passport.use(
       ...PASSPORT_FIELDS_ATTRIBUTES,
       passReqToCallback: true,
     },
-    function (req, username, password, done) {
-      mongodb
-        .collection('users')
-        .findOne({ uid: username })
-        .then(function (err, user) {
-          if (!user) {
-            return done(null, false, req.body)
-          }
-          return done(null, true, null)
-        })
+    async function (req, username, password, done) {
+      try {
+        const { appDb } = req.app.locals
+        const { mail } = req.body
+        const userAttributes = { $or: [{ uid: username }, { mail }] }
+        const user = await UserModel.findOne(appDb, userAttributes)
```

Comment:
do we need to make changes to this file? I thought we could only make changes in the `authController` since that's the only place a user signs up.

Reply:
Yes in this file the logic to query if a user is already registered, the done function passes the results to continue the registration to the controller.